src/economy.test.c

```diff
@@ -169,6 +170,77 @@ static void test_normals_recruit(CuTest * tc) {
     test_cleanup();
 }
 
+typedef struct request {
+    struct request *next;
+    struct unit *unit;
+    struct order *ord;
+    int qty;
+    int no;
+    union {
+        bool goblin;             /* stealing */
+        const struct luxury_type *ltype;    /* trading */
+    } type;
+} request;
+
+static void test_tax_cmd(CuTest *tc) {
+    order *ord;
+    faction *f;
+    region *r;
+    unit *u;
+    item_type *sword, *silver;
+    request *taxorders = 0;
+
+
+    test_cleanup();
+    set_param(&global.parameters, "taxing.perlevel", "20");
+    test_create_world();
+    f = test_create_faction(NULL);
+    r = findregion(0, 0);
+    assert(r && f);
+    u = test_create_unit(f, r);
+
+    ord = create_order(K_TAX, f->locale, "");
+    assert(ord);
+
+    tax_cmd(u, ord, &taxorders);
+    CuAssertPtrNotNull(tc, test_find_messagetype(u->faction->msgs, "error48"));
+    test_clear_messages(u->faction);
+
+    silver = get_resourcetype(R_SILVER)->itype;
+
+    sword = it_get_or_create(rt_get_or_create("sword"));
+    new_weapontype(sword, 0, 0.0, NULL, 0, 0, 0, SK_MELEE, 1);
+    i_change(&u->items, sword, 1);
+    set_level(u, SK_MELEE, 1);
+
+    tax_cmd(u, ord, &taxorders);
+    CuAssertPtrNotNull(tc, test_find_messagetype(u->faction->msgs, "error_no_tax_skill"));
+    test_clear_messages(u->faction);
+
+    set_level(u, SK_TAXING, 1);
+    tax_cmd(u, ord, &taxorders);
+    CuAssertPtrEquals(tc, 0, test_find_messagetype(u->faction->msgs, "error_no_tax_skill"));
+    CuAssertPtrNotNull(tc, taxorders);
+
+
+    r->land->money = 11;
+    expandtax(r, taxorders);
+    CuAssertPtrNotNull(tc, test_find_messagetype(u->faction->msgs, "income"));
+    CuAssertTrue(tc, i_get(u->items, silver) > 0 && i_get(u->items, silver) <= 11);
```

Comment:
Es gibt da eine merkwürdige Konstante, dass nur in Vielfachen von 10 verdient wird. Also verdient die Einheit hier 10, obwohl eigentlich 11 vorhanden wären... Man könnte sagen, der Test dokumentiert das immerhin. Aber man könnte auch gegen 10 testen.